http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ChannelPipelineInitializer.java

```diff
@@ -107,30 +114,39 @@ public void channelCreated(Channel ch) {
             }
         }
 
-        if (protocol == Protocol.HTTP2) {
-            configureHttp2(ch, pipeline);
-        } else {
-            configureHttp11(ch, pipeline);
-        }
+        configureProtocolHandlers(ch, pipeline, protocol);
+        configurePostProtocolHandlers(pipeline, protocol);
+    }
 
-        if (configuration.reapIdleConnections()) {
-            pipeline.addLast(new IdleConnectionReaperHandler(configuration.idleTimeoutMillis()));
+    private void configureProtocolHandlers(Channel ch, ChannelPipeline pipeline, Protocol protocol) {
+        switch (protocolNegotiation) {
```

Comment:
Can we add functional tests for those different combinations using a local mock server?

Reply:
Added NettyClientAlpnTest using MockH2Server from benchmarks